contracts/TuiChainLoanManagement.sol

```diff
@@ -0,0 +1,368 @@
+            _requestedValueDaiWei);
+
+        feeRecipient           = _feeRecipient;
+        loanRecipient          = _loanRecipient;
+        expirationTime         = block.timestamp.add(_secondsToExpiration);
+        fundingFeeDaiWeiPerDai = _fundingFeeDaiWeiPerDai;
+        paymentFeeDaiWeiPerDai = _paymentFeeDaiWeiPerDai;
+        requestedValueDai      = _requestedValueDai;
+
+        phase          = Phase.Funding;
+        loanToken      = TuiChainLoanToken(0);
+        fundedDai      = 0;
+        paidDai        = 0;
+        daiWeiPerToken = 0;
+    }
+
+    function _requirePositiveIntegerDai(
+        uint256 _daiWei) private pure returns (uint256 _dai)
+    {
+        require(
+            _daiWei > 0 && _daiWei.mod(1e18) == 0,
+            "Must be a positive, integer amount of Dai (multiple of 10^18)");
+
+        return _daiWei.div(1e18);
+    }
+
+    function _updatePhase(Phase _newPhase) private
+    {
+        assert(phase != _newPhase);
+
+        phase = _newPhase;
+        emit PhaseUpdate(_newPhase);
+    }
+
+    function _tryExpire() private
+    {
+        if (phase == Phase.Funding && block.timestamp >= expirationTime)
+            _updatePhase(Phase.Expired);
+    }
+
+    /**
+     * Expire the loan if the funding deadline has passed.
+     *
+     * Fails if current phase is not Funding or Expired.
+     *
+     * @return _expired true if the loan becomes or already was expired, false
+     *     otherwise
+     */
+    function checkExpiration() public returns (bool _expired)
+    {
+        _tryExpire();
+
+        require(phase == Phase.Funding || phase == Phase.Expired);
+
+        return phase == Phase.Expired;
+    }
+
+    /**
+     * Deposit some Dai to fund the loan.
+     *
+     * Fees are added to _valueDaiWei. The actual transferred value, in Dai Wei,
+     * is given by:
+     *
+     *     _valueDaiWei + fundingFeeDaiWeiPerDai * (_valueDaiWei / 1e18)
+     *
+     * @param _valueDaiWei Amount of funds to provide (before fees) in Dai Wei
+     */
+    function provideFunds(uint256 _valueDaiWei) external
+    {
+        _tryExpire();
+
+        // checks
+
+        require(phase == Phase.Funding);
+
+        uint256 valueDai = _requirePositiveIntegerDai(_valueDaiWei);
+        require(fundedDai.add(valueDai) <= requestedValueDai);
+
+        // effects
+
+        emit FundsUpdate(
+            fundedDai.mul(1e18), fundedDai.add(valueDai).mul(1e18));
+
+        fundedDai = fundedDai.add(valueDai);
+        unclaimedTokens[msg.sender] = unclaimedTokens[msg.sender].add(valueDai);
+
+        if (fundedDai == requestedValueDai)
+        {
+            _updatePhase(Phase.Active);
+
+            loanToken = new TuiChainLoanToken(requestedValueDai);
+        }
+
+        // interactions
+
+        uint256 feeDaiWei = fundingFeeDaiWeiPerDai.mul(valueDai);
+
+        dai.safeTransferFrom(
+            msg.sender, address(this), _valueDaiWei.add(feeDaiWei));
+
+        if (fundedDai == requestedValueDai)
```

Comment:
You have 2 if blocks with the same condition, i can see the objective of separating by effects and interactions but the code is quite simple. I don't think it's necessary to have 2 if blocks equal just for readability purposes. What do you think @joaop21

Reply:
I guess in this case it is largely inconsequential and mostly a matter of taste. Note though that the intent of the checks-effects-interactions pattern is to avoid reentrancy bugs, not improve readability, cf. https://fravoll.github.io/solidity-patterns/checks_effects_interactions.html.